src/main/java/com/stosik/parking/reservation/domain/evaluator/PriceCalculator.java

```diff
@@ -0,0 +1,19 @@
+package com.stosik.parking.reservation.domain.evaluator;
+
+import com.stosik.parking.reservation.domain.model.DriverType;
+import com.stosik.parking.reservation.domain.model.Reservation;
+
+import java.math.BigDecimal;
+
+public class PriceCalculator
+{
+    private final RateEvaluatorFactory rateEvaluatorFactory = new RateEvaluatorFactory();
```

Comment:
czemu tego nie wstrzykujesz?

Reply:
Wstzykiwanie wydałoby się tutaj odpowiednie, natomiast uznałem że bardziej optymalnym byłoby tworzenie jednej instacji i nią zarządzania bez potrzeby wyciągania za każdym razem gdy potrzebuje tego beana z kontenera.

Reply:
Ale przecież PriceCalculator też nie jest beanem.

Reply:
Skupilem sie na tym aby w kontenerze springowym byly tylko kluczowe obiekty, zauwazylem czesto ze ludzie na sile wciskaja i rejestruja beany po to tylko zeby miec swobodny dostep w kazdym miejscu w aplikacji do nich, natomiast podejscie ktore zaproponowalem opiera sie na koncepcji jednego punktu wejscia do modulu, gdzie naocznie widze co moja fasada potrzebuje i nie zasmiecam kontenera takimi beanami.